src/PetProject.Framework.Kafka/Consumer/Consumer.cs

```diff
@@ -111,15 +111,6 @@ public void Dispose()
             this.tokenSource?.Dispose();
         }
 
-        /// <inheritdoc />
-        /// <summary>
-        /// Decorator around Confluent Consumer to commit messages asynchronously after success.
-        /// </summary>
-        public Task<CommittedOffsets> CommitAsync()
```

Comment:
You shouldn't delete this. The user must be able to commit manually the messages.

You expose the method with the one you created or an overload. Otherwise the messages won't be committed if auto.commit property is disabled. It will only happen in the retries.

Reply:
Our consumer now commits messages by default if the handler doesn't throw an exception or if it throws an exception and the message was requeued. The way this consumer is built, it doesn't really make sense to expose a commit method. You would have to expose the offset properties or the internal Message objects which we are trying to avoid. We are trying to avoid exposing this low level stuff from the user.

Reply:
Exposing the CommitAsync without any parameter doesn't make any sense because it will commit every offset that was "polled" even if their processing hasn't been called/completed.

Reply:
Then you can expose one in which we send the message. Doing the manual commit after waiting for the success of the operation makes sense.

If you create an auto.commit but the processing of the message fails, you won't be able to recover because the consumer didn't fail, but the processing did,

Reply:
This "auto commit" feature that I implemented has nothing to do with the "auto.commit" property of kafka itself. This is "manually" commiting the message but it's abstracted from the user. If the processing fails, then commitasync won't be called.

Reply:
If it throws an exception. What if it doesn't throw one?

Reply:
Well if the handler delegate returns void the only way to signal an error is to throw an exception, surely the user would realize that. But I added a new Receive method that takes a delegate which returns a boolean so that this is a bit more clear.